### ISSUE TITLE
ArgumentError: invalid byte sequence in UTF-8 when sending ISO 8859-X encoded Mails

### ISSUE THREAD

**Issue by phil8900** [scott/helpy_imap#7]
Hey,
I encountered a strange error while using helpy_imap that occurs when users send ISO 8859-X encoded Mails. When fetching those Mails, an `<ArgumentError:` invalid byte sequence in UTF-8>` error pops up and the ticket is either empty or not showing up at all.
I have seen this error with Latin-1 (ISO 8859-1) and Latin-2 (ISO 8859-2) and therefore I suspect it might also occur for other versions of the ISO 8859 standard.

The odd thing is that when I was looking for that error I took a look at `entity.encoding.name` which is used in the `encode_entity(entity)` method and it was shown as ASCII-8BIT instead of ISO 8859-1 as I would have expected.

I'll investigate further :)

EDIT: I took a closer look, and I analyzed `@email.header.charset` and I got UTF-8 even for Latin-1/Latin-2 mails, so that didn't really help.

**Comment by phil8900** [comment]
As this issue was preventing us from using helpy_imap effectively, I implemented a workaround for this problem which leaves room for improvement I guess.
Most emails we get are already encoded in UTF-8, but there are some that are in ISO 8859-1 which is in line with the numbers here: https://w3techs.com/technologies/history_overview/character_encoding. (92% UTF-8, 3,9% ISO 8859-1, for websites). Based on that I assume that there won't be a significant loss by misrepresented characters when we first convert ASCII-8BIT messages to ISO-8859-1 which is an extension for US ASCII.  The downside might be that is not the most performant thing to do and therefore I would appreciate if someone has a smarter idea :)

**Comment by phil8900** [comment]
After some more testing I figured out that I made a mistake and that the proposed fix did not have the desired effect.
As I couldn't figure out why the "real" encoding of the @email variable does not represent the encoding of the raw email I came up with a different approach.
I added two checks if entity.force_encoding returns a String with valid encoding to distinguish UTF-8 and ISO-8859-X Strings, even if entity.encoding.name is just ASCII-8BIT which solved the problem for me.
